pkg/diagnostics/workflow_metrics.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diagnostics
+
+import (
+	"context"
+	diagUtils "github.com/dapr/dapr/pkg/diagnostics/utils"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	elapsedKey = tag.MustNewKey("elapsed")
+
+	reminderTypeKey = tag.MustNewKey("reminder_type")
+
+	executionTypeKey = tag.MustNewKey("execution_type")
+
+	isRetryableKey = tag.MustNewKey("is_retryable")
+)
+
+type workflowMetrics struct {
+	// Total successful workflow operation requests
+	workflowOperationsTotal *stats.Int64Measure
+	// Total failed workflow operation requests
+	workflowOperationsFailedTotal *stats.Int64Measure
```

Comment:
We can just add another key in `workflowOperationsCount` above to record Success/Failure Count. No need for another metric here. e.g. `processStatusKey` here: https://github.com/dapr/dapr/blob/474eab437329bccac9f5e37553413e103c46da84/pkg/diagnostics/component_monitoring.go#L173

Similar to this, we can add a `Status` Key with values as `Success`, `Retry` and `Failed`. That would also handle the usecase of `isRetryable` key being defined above